native/src/tablecloth.mli

```diff
@@ -1,10 +1,147 @@
+
+    {[List.map ~f:(Fun.constant 0) [1;2;3;4;5] = [0;0;0;0;0]]}
+
+    or {!Array.initialize}
+
+    {[Array.initialize ~length:6 ~f:(Fun.constant 0) = [|0;0;0;0;0;0|]]}
+  *)
+
+  val sequence : 'a -> 'b -> 'b
+  (** A function which always returns its second argument. *)
+
+  val flip : ('a -> 'b -> 'c) -> 'b -> 'a -> 'c
+  (** [flip f] reverses the argument order of the binary function [f].
+    For any arguments [x] and [y], [(flip f) x y] is [f y x].
+
+    Perhaps you want to [fold] something, but the arguments of a function you already have access to are in the wrong order.
+  *)
+
+  val apply : ('a -> 'b) -> 'a -> 'b
+  (** See {!Fun.(<|)} *)
+
+  val ( <| ) : ('a -> 'b) -> 'a -> 'b
+  (** [f <| x] is exactly the same as [f x].
+    It can help you avoid parentheses, which can be nice sometimes.
+    Maybe you want to apply a function to a [match] expression? That sort of thing.
+  *)
+
+  external pipe : 'a -> ('a -> 'b) -> 'b = "%revapply"
+  (** See {!Fun.(|>)} *)
+
+  external ( |> ) : 'a -> ('a -> 'b) -> 'b = "%revapply"
+  (** Saying [x |> f] is exactly the same as [f x], just a bit longer.
+
+  It is called the “pipe” operator because it lets you write “pipelined” code.
+  For example, say we have a [sanitize] function for turning user input into
+  integers:
+
+    
+    {[
+      (* Before *)
+      let sanitize (input: string) : int option =     
+        Int.fromString (String.trim input)
+    ]}
+
+  We can rewrite it like this:
+
+   {[
+    (* After *)
+    let sanitize (input: string) : int option =         
+      input
+      |> String.trim
+      |> Int.fromString
+    ]}
+
+  This can be overused! When you have three or four steps, the code often gets clearer if you break things out into
+  some smaller piplines assigned to variables. Now the transformation has a name, maybe it could have a type annotation.
+
+  It can often be more self-documenting that way!
+ *)
+
+  val compose : ('b -> 'c) -> ('a -> 'b) -> 'a -> 'c
+  (** Function composition, passing results along in the suggested direction. For
+example, the following code (in a very roundabout way) checks if a number divided by two is odd:
+
+    {[let isHalfOdd = Fun.(not << Int.isEven << Int.divide ~by:2)]}
+
+You can think of this operator as equivalent to the following:
+
+    {[(g << f)  ==  (fun x -> g (f x))]}
+
+So our example expands out to something like this:
+
+    {[let isHalfOdd = fun n -> not (Int.isEven (Int.divide ~by:2 n))]} *)
+
+  val ( << ) : ('b -> 'c) -> ('a -> 'b) -> 'a -> 'c
+  (** See {!Fun.compose} *)
+
+  val composeRight : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
+  (** Function composition, passing results along in the suggested direction. For
+example, the following code checks if the square root of a number is odd:
+
+    {[Int.squareRoot >> Int.isEven >> not]} *)
+
+  val ( >> ) : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
+  (** See {!Fun.composeRight} *)
+
+  val tap : 'a -> f:('a -> unit) -> 'a
```

Comment:
What does tap mean?

Reply:
It means "tap into"

Its based on:
- https://apidock.com/ruby/Object/tap
- https://lodash.com/docs/4.17.15#tap